ops/generate-daos-subgraph.js

```diff
@@ -4,14 +4,26 @@ const { migrationFileLocation: defaultMigrationFileLocation, network } = require
 const {   subgraphLocation: defaultSubgraphLocation } = require('./graph-cli')
 const path = require("path");
 const currentDir = path.resolve(`${__dirname}`)
+let ids = [];
 
 function daoYaml(contract, contractAddress, arcVersion) {
   const { abis, entities, eventHandlers } = yaml.safeLoad(
     fs.readFileSync(`${currentDir}/../src/mappings/${contract}/datasource.yaml`, "utf-8")
   );
+
+  let name = contract;
+  if (ids.indexOf(contract) == -1) {
+    ids.push(contract);
+  } else if (ids.indexOf(contract+contractAddress) == -1) {
+    ids.push(contract+contractAddress);
+    name = contract+contractAddress;
+  } else {
+    return;
+  }
```

Comment:
how about only  do `name = contract+ contractAddress` ?

Reply:
We can't because the graph node `codegen` function generates the `types` code based on the name, and we use this from our own code, so we must have at least one entity which has the original name.